src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int EMPTY_SIZE = 0;
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value, last, null);
+        if (last == null) {
+            first = node;
+        } else {
+            last.next = node;
+        }
+        last = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAddByIndex(index);
+        if (index >= size) {
+            add(value);
+        } else {
+            Node<T> current = getNodeByIndex(index);
+            Node<T> prev = current.prev;
+            Node<T> added = new Node<>(value, prev, current);
+            current.prev = added;
+            if (prev != null) {
+                prev.next = added;
+            } else {
+                first = added;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T elementFromList : list) {
+            add(elementFromList);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode = getNodeByIndex(index);
+        T oldValue = currentNode.item;
+        currentNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeToRemove = getNodeByIndex(index);
+        return unlink(nodeToRemove);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = searchNodeByItem(object);
+        if (node != null) {
+            unlink(node);
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == EMPTY_SIZE;
+    }
+
+    private void checkIndex(int index) {
```

Comment:
The method 'checkIndex' should be placed after all the public methods according to the best practices.